### PR TITLE
Add support for container app connected environment

### PR DESCRIPTION
**Pull request by ishank12**
### Issue describing the changes in this PR
Sync trigger api works when host is running on ContainerApp ManagedEnvironment.
However logic apps is running on ARC that has the concept of ConnectedEnvironment.
So adding the support of sync trigger for ConnectedEnvironment as well same as Managed Environment.

resolves #issue_for_this_pr

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script/Environment/EnvironmentSettingNames.cs
Comment (anandgmenon) [inline src/WebJobs.Script/Environment/EnvironmentSettingNames.cs:92]: Do we know if this env variable will actually exist in case of connected env?

**Review by anandgmenon — APPROVED** [review]

File: src/WebJobs.Script/Environment/EnvironmentSettingNames.cs
Comment (JatinSanghvi) [reply src/WebJobs.Script/Environment/EnvironmentSettingNames.cs:92]: Should we be more verbose with naming of this environment variable?

Comment (ishank12) [reply src/WebJobs.Script/Environment/EnvironmentSettingNames.cs:92]: We would need to add this from the dataplane.

Comment (ishank12) [reply src/WebJobs.Script/Environment/EnvironmentSettingNames.cs:92]: The name verbally seems confusing because there is Environment in the name which refers to infrastructure env of the containerApp.

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs
Comment (AmeyaRele) [inline src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: The IsAnyKubernetesEnvironment is used in some other places including for example in [ScriptHostIdProvider](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script/Host/ScriptHostIdProvider.cs#L73), [setting hostname](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L711), using [KubernetesEventGenerator](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L119), adding [EnvironmentReadyCheckMiddleware](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs#L41). Please ensure that these are paths that you would want for logic apps in ARC too.

Comment (ishank12) [reply src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: LogicApps specifically need it while setting hostname where FUNCTIONS_API_SERVER is set.
But looking it from infrastructure perpective, a connected environment in container apps is actually a Kubernetes Environment.

So its fine to consider that in IsAnyKubernetesEnvironment()

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: So it looks like your approach was to add this additional check 1:1 in every place where we were already checking for managed environment. Can we add a IsAnyContainerAppsEnvironment() or IsContainerAppsEnvironment() (whichever name makes more sense) helper that encapsulates these and use that everywhere. This'll be less error prone and easier to manage.

**Comment by mathewc** [comment]
Also, you need to fill out the PR template check boxes. E.g. does this need to be backported to inproc, etc.

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs
Comment (ishank12) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: fixed your suggestion. Thanks

**Review by mathewc — APPROVED** [review]

**Review by anandgmenon — APPROVED** [review]

**Review by anandgmenon — APPROVED** [review]

**Review by AmeyaRele — APPROVED** [review]

**Comment by kshyju** [comment]
@ishank12 Is this PR waiting on something before it can be merged? 
Also don't we need this change in the in-proc branch ?

**Pull request closed without merging**
